### DISCUSSION TITLE
Sample SCV Agent Transfer, How to add voicemail?

### DISCUSSION THREAD

**Discussion by SyndrixLane** [amazon-connect/amazon-connect-salesforce-scv#147] [Q&A] [answered]
Hello,

I recently setup SCV with partner telephony with aws

I am using the extension router package from here

We are using quick connects and the sample scv agent transfer contact flow

What I want to do is add the check staffing block and branch that not available into the voicemail function, but I am unsure how to check for the right agent, is the agent ID updated when the quick connect is selected? 

I guess I could go through my contact trace records for a call that has been transferred using the quick connect and see if the agent id or any of the contact data changes when the quick connect is used, this is a first attempt.

I am new to most of this but if someone could guide or point me in the right direction of how to accomplish this I am happy to build it.

Thanks for any help/advice

**Comment by dougjaso** [comment, marked as answer]
When you do an agent transfer via quick connect, the agent attribute is updated. If you look at the "Default agent transfer" contact flow, you can see how the agent data is accessed to set the agent personal queue. You can then check staffing using the agent data as referenced [here](https://docs.aws.amazon.com/connect/latest/adminguide/check-staffing.html).
